src/main/java/bio/terra/service/dataset/StorageDao.java

```diff
@@ -0,0 +1,114 @@
+        jdbcTemplate = new NamedParameterJdbcTemplate(jdbcConfiguration.getDataSource());
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
+    public List<StorageResource> getStorageResourcesByDatasetId(UUID datasetId) {
+        try {
+            MapSqlParameterSource params = new MapSqlParameterSource()
+                .addValue("dataset_id", datasetId);
+            return jdbcTemplate.query(SQL_GET, params, new StorageResourceMapper());
+        } catch (EmptyResultDataAccessException ex) {
+            throw new StorageResourceNotFoundException("Storage resource not found for dataset: "
+                + datasetId.toString());
+        }
+    }
+
+    public String getBucketStorageFromBucketResourceId(UUID bucketResourceId) {
+        try {
+            MapSqlParameterSource params = new MapSqlParameterSource()
+                .addValue("bucket_resource_id", bucketResourceId);
+            return Optional.ofNullable(jdbcTemplate.queryForObject(SQL_GET_BUCKET, params, new StorageResourceMapper()))
+                .map(StorageResource::getRegion).orElse(null);
+        } catch (EmptyResultDataAccessException ex) {
+            throw new StorageResourceNotFoundException("Storage resource not found for bucket: "
+                + bucketResourceId.toString());
+        }
+    }
+
+    private static class StorageResourceMapper implements RowMapper<StorageResource> {
+        public StorageResource mapRow(ResultSet rs, int rowNum) throws SQLException {
+            return new StorageResource()
+                .cloudPlatform(CloudPlatform.fromValue(rs.getString("cloud_platform")))
+                .cloudResource(rs.getString("cloud_resource"))
+                .region(rs.getString("region"));
+        }
+    }
+
+    @Transactional(propagation =  Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+    public void createStorageAttributes(List<StorageResource> storageResources, UUID datasetId) {
+        logger.debug("Create Operation: createStorageAttributes datasetId: {}", datasetId);
+
+        MapSqlParameterSource params = new MapSqlParameterSource()
+            .addValue("dataset_id", datasetId);
+
+        List<String> valuesList = new ArrayList<>();
+        for (StorageResource storageResource : storageResources) {
+            String regionParam = storageResource.getCloudResource() + "_region";
+            String cloudResourceParam = storageResource.getCloudResource() + "_cloudResource";
+            String platformParam = storageResource.getCloudResource() + "_cloudPlatform";
+            valuesList.add(String.format("(:dataset_id, :%s, :%s, :%s)",
+                regionParam, cloudResourceParam, platformParam));
+            params.addValue(regionParam, storageResource.getRegion());
+            params.addValue(cloudResourceParam, storageResource.getCloudResource());
+            params.addValue(platformParam, storageResource.getCloudPlatform().toString());
+        }
```

Comment:
I don't think this could ever happen. The `StorageResources` are instantiated from an enum, and so there would never be duplicate values. These values are created in memory and not passed in by the user.

Reply:
You're right - I was misunderstanding what the param names were getting set to.